modules/fundamental/src/ai/endpoints/mod.rs

```diff
@@ -123,3 +133,236 @@ pub async fn tool_call(
+    operation_id = "updateConversation",
+    params(
+        ("id", Path, description = "Opaque ID of the conversation")
+    ),
+    request_body = Conversation,
+    responses(
+        (status = 200, description = "The resulting conversation", body = Conversation),
+        (status = 400, description = "The request was invalid"),
+        (status = 404, description = "The AI service is not enabled or the conversation was not found")
+    )
+)]
+#[put("/v1/ai/conversations/{id}")]
+pub async fn update_conversation(
+    service: web::Data<AiService>,
+    db: web::Data<Database>,
+    id: web::Path<Uuid>,
+    user: UserDetails,
+    request: web::Json<Conversation>,
+    _: Require<Ai>,
+) -> actix_web::Result<impl Responder> {
+    let user_id = user.id;
+
+    let conversation_id = id.into_inner();
+
+    let (conversation, messages) = service
+        .upsert_conversation(
+            conversation_id,
+            user_id,
+            &request.messages,
+            request.seq,
+            db.as_ref(),
+        )
+        .await?;
+
+    let conversation = Conversation {
+        id: conversation.id,
+        updated_at: conversation.updated_at,
+        messages,
+        seq: conversation.seq,
+    };
+
+    Ok(HttpResponse::Ok().json(conversation))
+}
+
+#[utoipa::path(
+    tag = "ai",
+    operation_id = "listConversations",
+    params(
+        Query,
+        Paginated,
+    ),
+    responses(
+        (status = 200, description = "The resulting list of conversation summaries", body = PaginatedResults<ConversationSummary>),
+        (status = 404, description = "The AI service is not enabled")
+    )
+)]
+#[get("/v1/ai/conversations")]
+// Gets the list of the user's previous conversations
+pub async fn list_conversations(
+    service: web::Data<AiService>,
+    web::Query(search): web::Query<Query>,
+    web::Query(paginated): web::Query<Paginated>,
+    db: web::Data<Database>,
+    user: UserDetails,
+    _: Require<Ai>,
+) -> actix_web::Result<impl Responder> {
+    let user_id = user.id;
+
+    let result = service
+        .fetch_conversations(user_id, search, paginated, db.as_ref())
+        .await?;
+
+    let result = PaginatedResults {
+        items: result
+            .items
+            .into_iter()
+            .map(|c| ConversationSummary {
+                id: c.id,
+                summary: c.summary,
+                updated_at: c.updated_at,
+            })
+            .collect(),
+        total: result.total,
+    };
+
+    Ok(HttpResponse::Ok().json(result))
+}
+
+#[utoipa::path(
+    tag = "ai",
+    operation_id = "getConversation",
+    params(
+        ("id", Path, description = "Opaque ID of the conversation")
+    ),
+    responses(
+        (status = 200, description = "The resulting conversation", body = Conversation),
+        (status = 400, description = "The request was invalid"),
+        (status = 404, description = "The AI service is not enabled or the conversation was not found")
```

Comment:
It looks to me like there's an inconsistency in generating this. When the conversion is not found, it returns "ok, but emtpy" if the conversion is found, but doesn't belong to the user, it returns "not found".

Reply:
Ok, but in that case the part of "or the conversation was not found" is not true, is it?

Reply:
well yes.  It actually means that the conversation belongs to another user, but I'd rather lie than tell them that.

Reply:
But you're leaking that information anyway. Every time you get back a 404, it means "not allowed", or "other other".